src/actions/targetActions.js

```diff
@@ -45,3 +66,25 @@ export const addTarget = target =>
     }
   };
 
+export const selectTarget = target =>
+  (dispatch) => {
+    dispatch(deleteTargetSuccess({}, -1));
+    dispatch(target && target.id ? selectedTarget(target) : unSelectedTarget(target));
+  };
+
+export const newTarget = target =>
+  (dispatch) => {
+    dispatch(unSelectedTarget(target));
+    dispatch(addOrUpdateFutureTarget(target));
+  };
+
+export const removeTarget = (target, index) =>
+  async (dispatch) => {
+    try {
+      const targetResponse = await TargetApi.deleteTarget(target);
+      dispatch(deleteTargetSuccess(targetResponse, index));
+    } catch (err) {
+      dispatch(deleteTargetSuccess({}, index));
```

Comment:
I think that you should change function name because you're calling `deleteTargetSuccess` when the call to the API threw an error, it's confusing.